write.go

```diff
@@ -0,0 +1,95 @@
+package example
+
+import (
+	"io"
+	"io/ioutil"
+	"log"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/beyondstorage/go-storage/v4/pairs"
+	"github.com/beyondstorage/go-storage/v4/pkg/randbytes"
+	"github.com/beyondstorage/go-storage/v4/types"
+)
+
+func WriteData (store types.Storager, path string) {
+	// content to write
+	size := rand.Int63n(4 * 1024 * 1024)
+	r := io.LimitReader(randbytes.NewRand(), size)
+
+	// Write needs at least three arguments.
+	// `path` is the path of object.
+	// `r` is io.Reader instance for reading the data for uploading.
+	// `size` is the length, in bytes, of the data for uploading.
+	//
+	// Write will return two values.
+	// `n` is the size of write operation.
+	// `err` is the error during this operation.
+	n, err := store.Write(path, r, size)
+	if err != nil {
+		log.Fatalf("write %v: %v", path, err)
+	}
+
+	log.Printf("write size: %d", n)
+}
+
+func WriteWithCallback (store types.Storager, path string) {
+	// content to write
+	size := rand.Int63n(4 * 1024 * 1024)
+	r := io.LimitReader(randbytes.NewRand(), size)
+
+	cur := int64(0)
+	fn := func(bs []byte) {
+		cur += int64(len(bs))
+		log.Printf("write %d bytes already", cur)
+	}
+
+	// If IoCallback is specified, the storage will call it in every I/O operation.
+	// User could use this feature to implement progress bar.
+	n, err := store.Write(path, r, size, pairs.WithIoCallback(fn))
+	if err != nil {
+		log.Fatalf("write %v: %v", path, err)
+	}
+
+	log.Printf("write size: %d", n)
+}
+
+func WriteWithSignedURL (store types.Storager, path string, expire time.Duration) {
```

Comment:
Looks like our code not formatted.

Maybe we need to add format in `make build`: https://github.com/beyondstorage/go-storage-example/blob/master/Makefile#L22